_security-plugin/configuration/generate-certificates.md

```diff
@@ -99,26 +99,26 @@ If you already know the certificate details and don't want to specify them inter
 #!/bin/sh
 # Root CA
 openssl genrsa -out root-ca-key.pem 2048
-openssl req -new -x509 -sha256 -key root-ca-key.pem -subj "/C=CA/ST=ONTARIO/L=TORONTO/O=ORG/OU=UNIT/CN=ROOT" -out root-ca.pem -days 730
+openssl req -new -x509 -sha256 -key root-ca-key.pem -subj "/C=CA/ST=ONTARIO/L=TORONTO/O=ORG/OU=UNIT/CN=ROOT" -addext "subjectAltName = DNS:root.dns.a-record" -out root-ca.pem -days 730
 # Admin cert
 openssl genrsa -out admin-key-temp.pem 2048
 openssl pkcs8 -inform PEM -outform PEM -in admin-key-temp.pem -topk8 -nocrypt -v1 PBE-SHA1-3DES -out admin-key.pem
-openssl req -new -key admin-key.pem -subj "/C=CA/ST=ONTARIO/L=TORONTO/O=ORG/OU=UNIT/CN=ADMIN" -out admin.csr
+openssl req -new -key admin-key.pem -subj "/C=CA/ST=ONTARIO/L=TORONTO/O=ORG/OU=UNIT/CN=ADMIN" -addext "subjectAltName = DNS:admin.dns.a-record" -out admin.csr
```

Comment:
Not sure what's the right expectation when the CN is not a dns name. Should we make this a DNS name instead? @mpfz0r thoughts?

Reply:
For consistency, I'd use the same DNS name for the CN as for the subjectAltName. 
In practice, I think the CN will mostly be ignored if a subjectAltName is present.

Thank you for fixing this :pray:

Reply:
Thanks for the guidance @mpfz0r!  I will update accordingly.

On a side note, while I was testing this process today in my dev environment I was running into an error that `-addext "subjectAltName = DNS:admin.dns.a-record"` was not a valid option.  Any opposition to recommending that the user simply add the SAN info in an extension and then pass it to the `openssl` command with `-extfile`?  So the command to generate the signed cert would look something like:

`openssl x509 -req -in node.csr -CA ca.pem -CAkey ca.key -out node.crt -days 365 -sha256 -extfile node.ext`

I use my own root cert to sign my dev certs so I know the syntax is a little different than what we would use in the docs.

Reply:
@JeffH-AWS yeah that's quite a mess. see https://security.stackexchange.com/questions/74345/provide-subjectaltname-to-openssl-directly-on-the-command-line 
for all the gory details.

There's multiple solutions, but I think using `-extfile` is the easiest to understand.
Maybe add a line that generates it first?
e.g. `echo 'subjectAltName=DNS:admin.dns.a-record' > node.ext`

Reply:
Perfect!  I think that sounds good.  I was also considering that the docs advise use of a self-signed cert, although we could very easily add a couple steps for creating a root private key and cert, use the self-created root key/cert to sign all the requests for the cluster, and then add trust for the root cert to all the nodes and any clients that interact with the cluster.

Reply:
That may be a bit bigger of a lift which I can tackle in another issue.  For now I went ahead and applied the `-extfile` method for simplicity.